sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -386,6 +386,17 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val PARQUET_FILTER_PUSHDOWN_INFILTERTHRESHOLD =
+    buildConf("spark.sql.parquet.pushdown.inFilterThreshold")
+      .doc("The maximum number of values to filter push-down optimization for IN predicate. " +
+        "Large threshold won't necessarily provide much better performance. " +
+        "The experiment argued that 300 is the limit threshold. " +
+        "This configuration only has an effect when 'spark.sql.parquet.filterPushdown' is enabled.")
+      .internal()
+      .intConf
+      .checkValue(threshold => threshold > 0, "The threshold must be greater than 0.")
```

Comment:
Shell we allow, for example, `-1` here to disable this?

Reply:
```scala
case sources.In(name, values) if canMakeFilterOn(name) && shouldConvertInPredicate(name) 
  && values.distinct.length <= pushDownInFilterThreshold =>
```
How about `0`.  `values.distinct.length` will not be less than `0`.

Reply:
Let's use `-1`. Seems that's more consistent in the configurations.

Reply:
Yup.